nodebuilder/state/core.go

```diff
@@ -19,10 +19,15 @@ func coreAccessor(
 	signer *apptypes.KeyringSigner,
 	sync *sync.Syncer[*header.ExtendedHeader],
 	fraudServ libfraud.Service,
-) (*state.CoreAccessor, *modfraud.ServiceBreaker[*state.CoreAccessor]) {
+) (*state.CoreAccessor, Module, *modfraud.ServiceBreaker[*state.CoreAccessor]) {
+	if !corecfg.EndpointConfigured() {
+		log.Info("No core endpoint provided, running node without state access")
+		return nil, &stubbedStateModule{}, nil
+	}
```

Comment:
Can be two ProvideIfs, providing necessary coreaccessor basing on EndpointConfigured